packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -123,32 +125,52 @@ async function instantiateModule(
     }
   }
 
+  const ssrImportMeta = { url }
+  const ssrArguments: Record<string, any> = {
+    global: context.global,
+    [ssrModuleExportsKey]: ssrModule,
+    [ssrImportMetaKey]: ssrImportMeta,
+    [ssrImportKey]: ssrImport,
+    [ssrDynamicImportKey]: ssrDynamicImport,
+    [ssrExportAllKey]: ssrExportAll
+  }
+
+  let ssrModuleImpl = isProduction
+    ? result.code + `\n//# sourceURL=${mod.url}`
+    : `(0,function(${Object.keys(ssrArguments)}){\n${result.code}\n})`
+
+  const { map } = result
+  if (map?.mappings) {
+    if (mod.file) {
+      await injectSourcesContent(map, mod.file, moduleGraph)
```

Comment:
Honestly not sure if this call is needed in production. Someone needs to test with and without to see the difference. Otherwise, in development, this ensures that changes to local modules will not appear while debugging an earlier version of a module.